src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -154,3 +155,57 @@ def migrate_alert_rule(
         detector_workflow,
         workflow_data_condition_group,
     )
+
+
+def get_data_source(alert_rule: AlertRule) -> DataSource | None:
+    snuba_query = alert_rule.snuba_query
+    organization = alert_rule.organization
+    if not snuba_query or not organization:
+        # This shouldn't be possible, but just in case.
+        return None
+    try:
+        query_subscription = QuerySubscription.objects.get(snuba_query=snuba_query.id)
+    except QuerySubscription.DoesNotExist:
+        return None
+    try:
+        data_source = DataSource.objects.get(
+            organization=organization,
+            query_id=query_subscription.id,
+            type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION,
+        )
+    except DataSource.DoesNotExist:
+        return None
+    bulk_delete_snuba_subscriptions([query_subscription])
```

Comment:
it feels a bit strange to do this inside a function called `get_data_source` - could you instead just return the `DataSource` and then do the bulk delete in `dual_delete_migrated_alert_rule`?